metricproviders/prometheus/prometheus.go

```diff
@@ -140,12 +146,51 @@ func NewPrometheusProvider(api v1.API, logCtx log.Entry) *Provider {
 
 // NewPrometheusAPI generates a prometheus API from the metric configuration
 func NewPrometheusAPI(metric v1alpha1.Metric) (v1.API, error) {
+	secretKeys := []string{Address}
+	envValuesByKey := lookupKeysInEnv(secretKeys)
+	log.Infof("envValuesByKey: %v, len: %v", envValuesByKey, len(envValuesByKey))
+	if len(metric.Provider.Prometheus.Address) != 0 {
+		if !IsUrl(metric.Provider.Prometheus.Address) {
+			return nil, errors.New("prometheus address is not is url format")
+		}
+	} else if envValuesByKey[Address] != "" {
+		if IsUrl(envValuesByKey[Address]) {
+			metric.Provider.Prometheus.Address = envValuesByKey[Address]
+		} else {
+			return nil, errors.New("prometheus address is not is url format")
+		}
+	} else {
+		return nil, errors.New("prometheus address is not configured")
+	}
 	client, err := api.NewClient(api.Config{
 		Address: metric.Provider.Prometheus.Address,
 	})
 	if err != nil {
+		log.Errorf("Error in getting prometheus client: %v", err)
 		return nil, err
 	}
-
 	return v1.NewAPI(client), nil
 }
+
+func IsUrl(str string) bool {
+	u, err := url.Parse(str)
+	if err != nil {
+		log.Errorf("Error in parsing url: %v", err)
+	}
+	log.Infof("Parsed url: %v", u)
+	return err == nil && u.Scheme != "" && u.Host != ""
+}
+
+func lookupKeysInEnv(keys []string) map[string]string {
+	valuesByKey := make(map[string]string)
+	for i := range keys {
+		key := keys[i]
+		formattedKey := strings.ToUpper(strings.ReplaceAll(key, "-", "_"))
+		log.Infof("formattedKey: %v", formattedKey)
```

Comment:
Same as above. I feel that we should be a little conservative while using `info` level.